src/queue.ts

```diff
@@ -0,0 +1,41 @@
+import * as asyncLib from 'async';
+import * as fs from 'fs';
+import * as net from 'net';
+// will just write to wherever the process is running, but the server needs to be launched from the same directory so we use an abs path
+export const RESULTS_PATH = `${__dirname}/results.json`;
+
+function launchQueueServer() {
```

Comment:
i like this. curious if it works - have you tested the decompression script with this?
as discussed on call, if this doesn't work, I think you could make a bunch of unique results.json files (one for each analysis), and then append all of them in the end
Alternatively, could also make one results.json file for each item of the batch that's being run. I.e. if the code is run in batches of 13, then make 13 results.json files (results1.json, results2.json, etc) and then make sure each item writes to one of them, then append the 13 results.json files at the end. i think this actually might be the easiest thing to implement (if server doesn't work... i don't know much about the createServer function)

Reply:
what is even in results.json? I assumed there were issues if we did separate files bc we'd have to track weighted averages (which we don't rn) which is why I went with the queue process.

Reply:
objects that look like this (after running the script):
{
    "Number of games analyzed": 23,
    "maxKDRatio": 14,
    "pieceWithHighestKDRatio": [
        "K"
    ],
    "KDRatios": {
        "RA": 1.4,
        "NB": 0.8235294117647058,
        "BC": 0.7777777777777778,
        "Q": 1.8571428571428572,
        "K": 14,
        "BF": 0.8333333333333334,
        "NG": 1.4615384615384615,
        "RH": 1.0833333333333333,
        "PA": 0.25,
        "PB": 0.8333333333333334,
        "PC": 0.7142857142857143,
        "PD": 0.5714285714285714,
        "PE": 0.9285714285714286,
        "PF": 1.2,
        "PG": 0.42857142857142855,
        "PH": 0.2857142857142857,
        "ra": 1.4545454545454546,
        "nb": 0.8235294117647058,
        "bc": 1.2941176470588236,
        "q": 3.1,
        "k": 7.5,
        "bf": 0.6470588235294118,
        "ng": 1.4,
        "rh": 1,
        "pa": 0.8571428571428571,
        "pb": 0.6,
        "pc": 0.5,
        "pd": 0.5,
        "pe": 0.35714285714285715,
        "pf": 0.2222222222222222,
        "pg": 0.6666666666666666,
        "ph": 0
    },